src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php

```diff
@@ -46,16 +47,39 @@ public function __construct(ControllerNameParser $parser, LoggerInterface $logge
     }
 
     /**
-     * Loads a resource.
-     *
-     * @param mixed  $resource A resource
-     * @param string $type     The resource type
-     *
-     * @return RouteCollection A RouteCollection instance
+     * {@inheritdoc}
      */
     public function load($resource, $type = null)
     {
-        $collection = parent::load($resource, $type);
+        if (false !== $this->loading) {
+            // This can happen if a fatal error occurs in parent::load().
+            // Here is the scenario:
+            // - while routes are being loaded by parent::load() below, a fatal error
+            //   occurs (e.g. parse error in a controller while loading annotations);
```

Comment:
what about people not using the DelegatingLoader but a single loader (because they only use a single format for the routes) ?
this can't happen in the fullstack framework unless you redefine the route loader service definitions entirely (it always uses the delegating loader), but it can happen for people using the component standalone